shell/client/notifications-client.js

```diff
@@ -0,0 +1,229 @@
+                    devPackage, usage, globalDb.makeWildcardHost("static"));
+              }
+            }
+          } else {
+            const token = ApiTokens.findOne({grainId: row.grainId,
+                "owner.user.denormalizedGrainMetadata": {$exists: true}});
+            row.grainIcon = Identicon.iconSrcForDenormalizedGrainMetadata(
+                token.owner.user.denormalizedGrainMetadata, usage,
+                globalDb.makeWildcardHost("static"));
+          }
+        }
+      }
+
+      return row;
+    });
+  },
+});
+
+Template.notifications.helpers({
+  notificationCount: function () {
+    return Notifications.find({ userId: Meteor.userId(), isUnread: true }).count();
+  },
+});
+
+Template.notificationsPopup.events({
+  "click #notification-dropdown": function (event) {
+    return false;
+  },
+});
+
+Template.notificationItem.helpers({
+  isAppUpdates: function () {
+    return !!this.appUpdates;
+  },
+
+  hasIcon: function () {
+    return !!this.senderIcon || !!this.grainIcon;
+  },
+
+  notificationUrl: function () {
+    return getNotificationPath(this);
+  },
+
+  notificationTitle: function () {
+    if (this.admin) {
+      return "Notification from System";
+    } else if (this.appUpdates) {
+      return "App updates are available";
+    } else if (this.ongoing) {
+      return this.grainTitle + " is backgrounded";
+    } else {
+      return false;
+    }
+  },
+
+  titleHelperText: function () {
+    if (this.admin) {
+      return "Dismiss this system notification";
+    } else if (this.referral) {
+      return "Dismiss this referral notification";
+    } else if (this.ongoing) {
+      return "Stop the background app";
+    } else {
+      return "Dismiss this notification";
+    }
+  },
+
+  dismissText: function () {
+    if (this.admin && this.admin.type === "reportStats") {
+      return false;
+    } else if (this.ongoing) {
+      return "Cancel";
+    } else {
+      return "Dismiss";
+    }
+  },
+
+  adminLink: function () {
+    return this.admin && this.admin.action;
+  },
+
+  appUpdatesList: function () {
+    return _.values(this.appUpdates);
+  },
+
+  paidUser: function () {
+    const plan = Meteor.user().plan;
+    return plan && plan !== "free";
+  },
+});
+
+Template.notificationItem.events({
+  "click .cancel-notification": function (event) {
+    Meteor.call("dismissNotification", this._id);
+    return false;
+  },
+
+  "click .accept-notification": function (event) {
+    if (this.appUpdates) {
+      Meteor.call("updateApps", this.appUpdates, (err) => {
+        // TODO(someday): if (err)
+        Meteor.call("dismissNotification", this._id);
+      });
+    }
+
+    return false;
+  },
+
+  "click .dismiss-notification": function (event) {
+    Meteor.call("dismissNotification", this._id);
+  },
+});
+
+Meteor.startup(function () {
+  Meteor.subscribe("notifications");
+
+  Meteor.autorun(function () {
+    Meteor.subscribe("notificationGrains",
```

Comment:
Consider only calling this once the "notifications" subscription is `.ready()` - should save some sub/unsub/resub round trips on initial pageload if you have multiple notifications.

Otherwise, as soon as each `Notification` filters in, you're cancelling the previous sub and rerunning this subscription with a different set of arguments.